a3p-integration/proposals/z:acceptance/test-lib/psm-lib.js

```diff
@@ -0,0 +1,513 @@
+
+/**
+ * @param {string} anchor
+ */
+export const getPsmGovernance = async anchor => {
+  const governanceRaw = await agoric.follow(
+    '-lF',
+    `:published.psm.IST.${anchor}.governance`,
+    '-o',
+    'text',
+  );
+  const { current } = marshaller.fromCapData(JSON.parse(governanceRaw));
+  return current;
+};
+
+/**
+ * @param {string} anchor
+ */
+export const getPsmMetrics = async anchor => {
+  const metricsRaw = await agoric.follow(
+    '-lF',
+    `:published.psm.IST.${anchor}.metrics`,
+    '-o',
+    'text',
+  );
+
+  return marshaller.fromCapData(JSON.parse(metricsRaw));
+};
+
+export const checkGovParams = async (t, expected, psmName) => {
+  const current = await getPsmGovernance(psmName);
+
+  t.log({
+    give: current.WantMintedFee.value,
+    want: current.GiveMintedFee.value,
+    mintLimit: current.MintLimit,
+  });
+
+  t.like(current, expected);
+};
+
+/**
+ *
+ * @param {string} name
+ * @param {{
+ *   denom: string,
+ *   value: string
+ * }} fund
+ */
+export const initializeNewUser = async (name, fund) => {
+  const psmTrader = await addUser(name);
+  await Promise.all([
+    bankSend(psmTrader, `20000000ubld,${fund.value}${fund.denom}`),
+    bankSend(psmTrader, `1000000uist`, GOV1ADDR),
+  ]);
+};
+
+/**
+ *
+ * @param {string} userName
+ * @param {{
+ *   denom: string,
+ *   value: string
+ * }} expectedAnchorFunds
+ */
+export const checkUserInitializedSuccessfully = async (
+  userName,
+  expectedAnchorFunds,
+) => {
+  const userAddress = await getUser(userName);
+
+  const balance = await getBalances([userAddress], expectedAnchorFunds.denom);
+  assert(balance >= BigInt(expectedAnchorFunds.value));
+};
+
+/**
+ * Similar to https://github.com/Agoric/agoric-3-proposals/blob/422b163fecfcf025d53431caebf6d476778b5db3/packages/synthetic-chain/src/lib/commonUpgradeHelpers.ts#L123-L139
+ * However, in the case where "address" is not provisioned "agoric wallet send" is needed because
+ * "agops perf satisfaction" tries to follow ":published.wallet.${address}" which blocks the execution because no such path exists in
+ * vstorage. In situations like this "agoric wallet send" seems a better choice as it doesn't depend on following user's vstorage wallet path
+ *
+ * @param {string} address
+ * @param {Promise<string>} offerPromise
+ */
+export const sendOfferAgoric = async (address, offerPromise) => {
+  const offerPath = await mkTemp('agops.XXX');
+  const offer = await offerPromise;
+  await fsp.writeFile(offerPath, offer);
+
+  await agoric.wallet(
+    '--keyring-backend=test',
+    'send',
+    '--offer',
+    offerPath,
+    '--from',
+    address,
+  );
+};
+
+/**
+ * @param {string} address
+ * @param {Array<any>} params
+ */
+export const agopsPsm = (address, params) => {
+  const newParams = ['psm', ...params];
+  const offerPromise = executeCommand(agopsLocation, newParams);
+  return sendOfferAgoric(address, offerPromise);
+};
+
+/**
+ *
+ * @param {number} base
+ * @param {number} fee
+ * @returns
```

Comment:
I suspect this makes `yarn lint` fail, so it needs to be cleaned up before the PR is green.

Reply:
Resolved in https://github.com/Agoric/agoric-sdk/pull/10273/commits/11b336bb2c2fe58adc25a7a779ec6996d23e7d10